src/promptflow/promptflow/executor/_base.py

```diff
@@ -0,0 +1,70 @@
+import abc
+from pathlib import Path
+from typing import Any, Callable, Dict, Mapping, Optional
+
+from promptflow._constants import LINE_TIMEOUT_SEC
+from promptflow.executor._flow_nodes_scheduler import DEFAULT_CONCURRENCY_FLOW
+from promptflow.executor._result import AggregationResult, LineResult
+from promptflow.storage import AbstractRunStorage
+
+
```

Comment:
We expose ExecutorProxy instead of executor here, since the code in python is just a proxy.

Reply:
See this PR: https://github.com/microsoft/promptflow/pull/1179

Reply:
hmm, let me double confirm. I saw a `PythonExecutorProxy` whose `execute_line_async` is not implemented. What's your plan: 
1. implement `execute_line_async` for python proxy and SDK team will refactor `Submitter` to avoid using `FlowExecutor` directly
2. make `PythonExecutorProxy` an `APIBasedExecutorProxy` and it would be used for batch engine only
3. make `PythonExecutorProxy` an `APIBasedExecutorProxy` and finally we will start a service for even 1 line execution

Reply:
As offline synced, I will merge a `CsharpExecutoreProxy` using `pythonnet` for the initial version and will discuss whether we should keep it in following discussions